jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/contrastrules/library/LibraryVulnerabilityAssessmentPolicy.java

```diff
@@ -0,0 +1,139 @@
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.oauth2.sdk.util.MapUtils;
+import com.tmobile.cloud.awsrules.utils.PacmanUtils;
+import com.tmobile.cloud.constants.PacmanRuleConstants;
+import com.tmobile.cloud.contrastrules.utils.ResourceVulnerability;
+import com.tmobile.cloud.model.CveDetails;
+import com.tmobile.cloud.model.VulnerabilityInfo;
+import com.tmobile.pacman.commons.PacmanSdkConstants;
+import com.tmobile.pacman.commons.exception.InvalidInputException;
+import com.tmobile.pacman.commons.policy.Annotation;
+import com.tmobile.pacman.commons.policy.BasePolicy;
+import com.tmobile.pacman.commons.policy.PacmanPolicy;
+import com.tmobile.pacman.commons.policy.PolicyResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@PacmanPolicy(key = "check-vulnerability-exists-for-library", desc = "Checks for the presence of vulnerabilities.", severity = PacmanSdkConstants.SEV_HIGH, category = PacmanSdkConstants.SECURITY)
+public class LibraryVulnerabilityAssessmentPolicy extends BasePolicy {
+
+    private static final Logger logger = LoggerFactory.getLogger(LibraryVulnerabilityAssessmentPolicy.class);
+
+    private static final String CONTRAST_URL_TEMPLATE = "https://eval.contrastsecurity.com/Contrast/static/ng/" +
+            "index.html#/%s/libraries?quickFilter=VULNERABLE&filterText=%s";
+
+    /**
+     * @param policyParam        parameters for policy validation
+     * @param resourceAttributes this is a resource in context which needs to be scanned this is provided by execution engine
+     */
+    @Override
+    public PolicyResult execute(final Map<String, String> policyParam, Map<String, String> resourceAttributes) {
+        logger.debug("LibraryVulnerabilityAssessmentPolicy started");
+        if (MapUtils.isNotEmpty(policyParam) && !PacmanUtils.doesAllHaveValue(policyParam.get(PacmanRuleConstants.SEVERITY),
+                policyParam.get(PacmanRuleConstants.CATEGORY))) {
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+        String policyId = policyParam.get(PacmanSdkConstants.POLICY_ID);
+        String accountId = resourceAttributes.get(PacmanRuleConstants.ACCOUNTID);
+        MDC.put("executionId", policyParam.get("executionId"));
+        MDC.put("policyId", policyId);
+        String vulnerabilities;
+        String severity = policyParam.get("severityToCheck");
+        List<VulnerabilityInfo> vulnerabilityDetailsList = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            vulnerabilities = resourceAttributes.get("vulns");
```

Comment:
what is `resourceAttributes` is empty? I saw such check in many policies. Is it possible?

Reply:
yes, Updated